lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1368,7 +1409,12 @@ SPIRVValue *LLVMToSPIRV::transValueWithoutDecoration(Value *V,
     if (ST->isAtomic())
       return transAtomicStore(ST, BB);
 
-    std::vector<SPIRVWord> MemoryAccess(1, 0);
+    // Keep this vector to store MemoryAccess operands for both Alignment and
+    // Aliasing information. It's quite incorrect, since the first is of a
+    // SPIRVWord type, while the second one is of SPIRVId type, but creation
+    // of the second vector would make fulfilling '3.26 Memory Operands'
+    // requirements about operands order a bit tricky.
```

Comment:
SPIRV id is the same SPIRV word. See https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#Id and https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/56988bb67213554db07932d0f0765a5f0de2e581/lib/SPIRV/libSPIRV/SPIRVEnum.h#L54-L55

Reply:
I meant to say, that SPIRVId always points to some object in SPIR-V module (instruction, variable etc), whilst SPIRVWord can also be a constant literal. But I actually missed "An <id> always consumes one word". Removed this comment.